core/src/main/java/org/opensearch/sql/expression/datetime/DateTimeFunction.java

```diff
@@ -101,6 +103,30 @@ public class DateTimeFunction {
   // Mode used for week/week_of_year function by default when no argument is provided
   private static final ExprIntegerValue DEFAULT_WEEK_OF_YEAR_MODE = new ExprIntegerValue(0);
 
+  // Map used to determine format output for the extract function
+  private static final Map<String, String> extract_formats =
+      ImmutableMap.<String, String>builder()
+          .put("MICROSECOND", "%f")
+          .put("SECOND", "%s")
+          .put("MINUTE", "%i")
+          .put("HOUR", "%H")
+          .put("DAY", "%d")
+          .put("WEEK", "%V")
+          .put("MONTH", "%m")
```

Comment:
QUARTER (%Q)?

Reply:
Logic would be easier than calling another function only for quarter which doing some magic calculations.

Reply:
I had to do this because these utilise the `DATE_FORMAT` function, but "%Q" does not seem to exist. I'll try to work around it to directly use the Java `DateTimeFormatter` if that's preferred.

Reply:
So you can add %Q to the handlers list

Reply:
Is that ok to do? I assumed it wasn't since it is not behaviour listed for MySQL's `DATE_FORMAT` function. [DATE_FORMAT(date,format)](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-functions.html#function_date-format).

If this is ok then I'll just add it to the handlers so that it is cleaner.

Reply:
Right... Dilemma.

Reply:
You can create a new handler list which would convert units (`DAY`, `MINUTE_SECOND`, etc) into java time formatters like `DD`, `mm:ss` and so. Currently you convert units into SQL formatters (ones with `%` sign) and then into java formatters. You can omit this step and freely add `QUARTER` here too.

Reply:
Ah I see. Yes that makes more sense. Thanks!